Packs/PhishingURL/Scripts/DBotPredictURLPhishing/DBotPredictURLPhishing.py

```diff
@@ -0,0 +1,705 @@
+    return list(set(json.loads(res[0]["Contents"]).get("URL", [])))
+
+
+def load_demisto_model():
+    model_64_str = get_model_data(URL_PHISHING_MODEL_NAME)[0]
+    model = decode_model_data(model_64_str)
+    return model
+
+def get_final_urls(urls, max_urls, model):
+    final_url = []
+    seen = []
+    low_priority_urls = []
+    i = 0
+    for url in urls:
+        if i < max_urls:
+            if extract_domainv2(url) in seen or extract_domainv2(url) in model.top_domains:
+                low_priority_urls.append(url)
+            else:
+                final_url.append(url)
+                seen.append(extract_domainv2(url))
+                i += 1
+    if len(final_url) < max_urls:
+        final_url = final_url + low_priority_urls[:min(len(low_priority_urls), max_urls-len(final_url))]
+    return final_url
+
+
+def extract_embedded_urls_from_html(html):
+    embedded_urls = []
+    soup = BeautifulSoup(html)
+    for a in soup.findAll('a'):
+        if a.has_attr('href'):
+            if a['href'] not in a.get_text():
+                embedded_urls.append(a['href'])
+    return embedded_urls
+
+
+def main():
+    msg_list = []
+    exist, demisto_major_version, demisto_minor_version = oob_model_exists_and_updated()
+    reset_model = demisto.args().get('resetModel', 'False') == 'True'
+    debug = demisto.args().get('debug', 'False') == 'True'
+    force_model = demisto.args().get('forceModel', 'False') == 'True'
+    email_body = demisto.args().get('emailBody',"")
+    email_html = demisto.args().get('emailHTML', "")
+    max_urls= int(demisto.args().get('maxNumberOfURL', 5))
+    if debug:
+        if exist:
+            msg_list.append(MSG_MODEL_VERSION_IN_DEMISTO % (demisto_major_version, demisto_minor_version))
+        else:
+            msg_list.append(MSG_NO_MODEL_IN_DEMISTO)
+    if reset_model or not exist or (
+            demisto_major_version < MAJOR_VERSION and demisto_minor_version == MINOR_DEFAULT_VERSION):
+        msg_list.append(load_oob_model(OUT_OF_THE_BOX_MODEL_PATH))
+
+    elif (demisto_major_version == MAJOR_VERSION):
+        msg_list.append(MSG_NO_ACTION_ON_MODEL)
+    elif (demisto_major_version < MAJOR_VERSION) and (demisto_minor_version > MINOR_DEFAULT_VERSION):
+        model_docker = load_model_from_docker()
+        model_docker_minor = model_docker.minor
+        model = load_demisto_model()
+        model_docker.logos_dict = model.logos_dict
+        model_docker.clf.named_steps.preprocessor.named_transformers_[
+            'image'].named_steps.trans.logo_dict = model.logos_dict
+        model_docker.minor += 1
+        save_model_in_demisto(model_docker)
+        msg_list.append(MSG_UPDATE_LOGO % (MAJOR_VERSION, model_docker_minor, model.major, model.minor))
+    else:
+        msg_list.append(MSG_WRONG_CONFIG_MODEL)
+        return_error(MSG_WRONG_CONFIG_MODEL)
+    model_64_str = get_model_data(URL_PHISHING_MODEL_NAME)[0]
+    model = decode_model_data(model_64_str)
+    if email_body:
```

Comment:
Please do this step first, the URL extraction - so you will know if there is no URL URL to process - you don't even to load the model....

Reply:
Not possible since the URL extraction will check if some URL belongs to whitelist and the whitelist is inside the model itself.